src/inet/IPEndPointBasis.cpp

```diff
@@ -1292,20 +1293,39 @@ void IPEndPointBasis::GetPacketInfo(nw_connection_t aConnection, IPPacketInfo &
     aPacketInfo.DestPort    = nw_endpoint_get_port(dest_endpoint);
 }
 
-INET_ERROR IPEndPointBasis::GetEndPoint(nw_endpoint_t & aEndPoint, const IPAddress aAddress, uint16_t aPort)
+INET_ERROR IPEndPointBasis::GetEndPoint(nw_endpoint_t & aEndPoint, const IPAddressType aAddressType, const IPAddress aAddress,
+                                        uint16_t aPort)
 {
     INET_ERROR res = INET_NO_ERROR;
-
+    IPAddress addr;
```

Comment:
This seems redundant.

Reply:
It looks redundant but it was not perfectly redundant. But I agree I could have make the code clearer and I could have added a comment to explains the rationale here. Thanks to remembering me to not be lazy !

So the story behind this code is that the endpoint is an IPv4 endpoint and during a new connection the NW.framework want the IP as a string. BUT https://github.com/project-chip/connectedhomeip/blob/e0aa5e5102d12150c0e17868579ddf43aec59884/src/inet/IPAddress-StringFuncts.cpp#L62 converts Any address ("::") to the IPv6 format by default if the address is of type Any, while we expect the IPv4 Any ("0.0.0.0").

So I have modified the code to be clearer about this particular case and I have added a comment. Thanks again.